csharp/src/Apache.Arrow/Arrays/BinaryArray.cs

```diff
@@ -171,13 +171,22 @@ public TBuilder Clear()
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public int GetValueOffset(int index)
         {
+            if (index < 0 || index >= Length)
```

Comment:
Getting the value offset at `index == Length` is valid, right? That's how you can tell how long the last element is.

Reply:
Indeed, I fixed it.

Reply:
> Getting the value offset at index == Length is valid, right?

I think that this is invalid. Because it returns the offset of nonexistent value.

> That's how you can tell how long the last element is.

I think that users should use `GetValueLength()` instead of computing value length by `GetValueOffset()` values. If advanced users want to compute by themselves, using `ValueOffsets` directly may be better.

Reply:
The more I look at this method, the less value I see in it. Maybe we should just remove the method all together and people can index into ValueOffsets directly. That’s all this method does.

Reply:
I agree with removing this method.

Reply:
I agree with removing but I'm a little concerned about the impact on current users.
If we don't need to consider it, I'll remove it.

Reply:
How about marking this method "deprecated" in the next release, then remove this method in the future release?
(I'm not sure how to do this in C#. Documentation and warning message?)

Reply:
In .NET there is an [Obsolete] attribute you can apply to a method. Callers will receive a build time warning that the method is obsolete. We can add that.

Reply:
Okey. I'll add an [Obsolete] attribute.